src/Microsoft.PowerShell.Security/security/CertificateProvider.cs

```diff
@@ -3340,17 +3340,25 @@ private DnsNameRepresentation GetDnsNameRepresentation(string dnsName)
         /// </summary>
         public DnsNameProperty(X509Certificate2 cert)
         {
-            _dnsList = new List<DnsNameRepresentation>();
-
-            // extract DNS name from subject distinguish name
-            // if it exists and does not contain a comma
-            // a comma, indicates it is not a DNS name
-            if (cert.Subject.StartsWith(distinguishedNamePrefix, StringComparison.OrdinalIgnoreCase) &&
-                !cert.Subject.Contains(','))
+            // Extract DNS name from Subject distinguished name
+            foreach (X500RelativeDistinguishedName rdn in cert.SubjectName.EnumerateRelativeDistinguishedNames())
             {
-                string parsedSubjectDistinguishedDnsName = cert.Subject.Substring(distinguishedNamePrefix.Length);
-                DnsNameRepresentation dnsName = GetDnsNameRepresentation(parsedSubjectDistinguishedDnsName);
-                _dnsList.Add(dnsName);
+                // Check distinguished name has single element and has Common Name (CN) OID
+                // We skip multi value CN-ID RDNs since in RFC it is not clear how they are handled
+                // Given we have cases where Subject field is duplicated in SAN, we continue to add entries to DNS name list
+                if (!rdn.HasMultipleElements)
+                {
+                    string oid = rdn.GetSingleElementType().Value;
+                    string rdnValue = rdn.GetSingleElementValue();
+
+                    if (!string.IsNullOrEmpty(oid) &&
+                        !string.IsNullOrEmpty(rdnValue) &&
+                        oid.Equals(CommonNameOid, StringComparison.Ordinal))
```

Comment:
I wonder why do we check `oid` if CommonNameOid is never null:
```suggestion
                    if (!string.IsNullOrEmpty(rdnValue) &&
                        CommonNameOid.Equals(oid, StringComparison.Ordinal))
```

Reply:
Fixed in e00c0b5fdb7b036d0d32b03c29b0419b7efb02c3